diff -ruN o/main.c pc/main.c
--- o/main.c	2020-08-17 23:42:27.000000000 +0300
+++ pc/main.c	2021-05-07 15:07:57.542852246 +0300
@@ -133,7 +133,7 @@
 
 	if (stb_type == UNKNOWN)
 	{
-		FILE *file = fopen("/proc/stb/info/chipset", "r");
+		FILE *file = fopen("/usr/local/e2/etc/info/chipset", "r");
 		if (file)
 		{
 			char buf[32];
@@ -306,7 +306,7 @@
 
 	if (stb_type == UNKNOWN)
 	{
-		FILE *file = fopen("/proc/stb/info/model", "r");
+		FILE *file = fopen("/usr/local/e2/etc/info/model", "r");
 		if (file)
 		{
 			char buf[32];
@@ -546,35 +546,35 @@
 	}
 	else
 	{
-		fp = fopen("/proc/stb/vmpeg/0/aspect", "r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/aspect", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&aspect);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_width", "r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/dst_width", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&dst_width);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_height", "r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/dst_height", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&dst_height);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_top", "r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/dst_top", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&dst_top);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_left", "r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/dst_left", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
@@ -925,7 +925,7 @@
 
 		munmap((void*)data, 100);
 
-		fp=fopen("/proc/stb/vmpeg/0/yres","r");
+		fp=fopen("/usr/local/e2/etc/vmpeg/0/yres","r");
 		while (fgets(buf,sizeof(buf),fp))
 			sscanf(buf,"%x",&res);
 		fclose(fp);
@@ -1073,7 +1073,7 @@
 	else if (stb_type == XILLEON)
 	{
 		// grab xilleon pic from decoder memory
-		fp = fopen("/proc/stb/vmpeg/0/xres","r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/xres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
@@ -1082,7 +1082,7 @@
 			}
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/yres","r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/yres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
@@ -1276,7 +1276,7 @@
 		*xres=0;
 		*yres=0;
 
-		fp = fopen("/proc/stb/vmpeg/0/xres","r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/xres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
@@ -1286,7 +1286,7 @@
 			fclose(fp);
 		}
 
-		fp = fopen("/proc/stb/vmpeg/0/yres","r");
+		fp = fopen("/usr/local/e2/etc/vmpeg/0/yres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
