diff -ruN b/main.c a/main.c
--- b/main.c	2019-01-24 14:25:34.002622872 +0300
+++ a/main.c	2019-01-24 14:27:51.551913925 +0300
@@ -133,7 +133,7 @@
 
 	if (stb_type == UNKNOWN)
 	{
-		FILE *file = fopen("/proc/stb/info/chipset", "r");
+		FILE *file = fopen("/usr/local/e2/etc/stb/info/chipset", "r");
 		if (file)
 		{
 			char buf[32];
@@ -301,7 +301,7 @@
 
 	if (stb_type == UNKNOWN)
 	{
-		FILE *file = fopen("/proc/stb/info/model", "r");
+		FILE *file = fopen("/usr/local/e2/etc/stb/info/model", "r");
 		if (file)
 		{
 			char buf[32];
@@ -541,35 +541,35 @@
 	}
 	else
 	{
-		fp = fopen("/proc/stb/vmpeg/0/aspect", "r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/aspect", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&aspect);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_width", "r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/dst_width", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&dst_width);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_height", "r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/dst_height", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&dst_height);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_top", "r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/dst_top", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
 				sscanf(buf,"%x",&dst_top);
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/dst_left", "r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/dst_left", "r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf), fp))
@@ -920,7 +920,7 @@
 
 		munmap((void*)data, 100);
 
-		fp=fopen("/proc/stb/vmpeg/0/yres","r");
+		fp=fopen("/usr/local/e2/etc/stb/vmpeg/0/yres","r");
 		while (fgets(buf,sizeof(buf),fp))
 			sscanf(buf,"%x",&res);
 		fclose(fp);
@@ -1068,7 +1068,7 @@
 	else if (stb_type == XILLEON)
 	{
 		// grab xilleon pic from decoder memory
-		fp = fopen("/proc/stb/vmpeg/0/xres","r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/xres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
@@ -1077,7 +1077,7 @@
 			}
 			fclose(fp);
 		}
-		fp = fopen("/proc/stb/vmpeg/0/yres","r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/yres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
@@ -1271,7 +1271,7 @@
 		*xres=0;
 		*yres=0;
 
-		fp = fopen("/proc/stb/vmpeg/0/xres","r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/xres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
@@ -1281,7 +1281,7 @@
 			fclose(fp);
 		}
 
-		fp = fopen("/proc/stb/vmpeg/0/yres","r");
+		fp = fopen("/usr/local/e2/etc/stb/vmpeg/0/yres","r");
 		if (fp)
 		{
 			while (fgets(buf,sizeof(buf),fp))
