diff -ruN b/gstdtsdownmix.c a/gstdtsdownmix.c
--- b/gstdtsdownmix.c	2017-12-08 17:28:59.000000000 +0300
+++ a/gstdtsdownmix.c	2018-05-27 13:24:44.000000000 +0300
@@ -388,13 +388,13 @@
 static void set_stcmode(GstDtsDownmix *dts)
 {
 	FILE *f;
-	f = fopen("/proc/stb/stc/0/sync", "r");
+	f = fopen("/usr/local/e2/etc/stb/stc/0/sync", "r");
 	if (f)
 	{
 		fgets(dts->stcmode, sizeof(dts->stcmode), f);
 		fclose(f);
 	}
-	f = fopen("/proc/stb/stc/0/sync", "w");
+	f = fopen("/usr/local/e2/etc/stb/stc/0/sync", "w");
 	if (f)
 	{
 		fprintf(f, "audio");
@@ -406,7 +406,7 @@
 {
 	if (dts->stcmode[0])
 	{
-		FILE *f = fopen("/proc/stb/stc/0/sync", "w");
+		FILE *f = fopen("/usr/local/e2/etc/stb/stc/0/sync", "w");
 		if (f)
 		{
 			fputs(dts->stcmode, f);
@@ -419,7 +419,7 @@
 {
 	FILE *f;
 	char buffer[32] = {0};
-	f = fopen("/proc/stb/audio/ac3", "r");
+	f = fopen("/usr/local/e2/etc/stb/audio/ac3", "r");
 	if (f)
 	{
 		fread(buffer, sizeof(buffer), 1, f);
diff -ruN b/gstdvbaudiosink.c a/gstdvbaudiosink.c
--- b/gstdvbaudiosink.c	2017-12-08 17:28:59.000000000 +0300
+++ a/gstdvbaudiosink.c	2018-05-27 13:24:36.000000000 +0300
@@ -349,7 +349,7 @@
 {
 	FILE *f;
 	char buffer[32] = {0};
-	f = fopen("/proc/stb/audio/ac3", "r");
+	f = fopen("/usr/local/e2/etc/stb/audio/ac3", "r");
 	if (f)
 	{
 		fread(buffer, sizeof(buffer), 1, f);
diff -ruN b/gstdvbvideosink.c a/gstdvbvideosink.c
--- b/gstdvbvideosink.c	2017-12-08 17:28:59.000000000 +0300
+++ a/gstdvbvideosink.c	2018-05-27 13:24:23.000000000 +0300
@@ -1611,7 +1611,7 @@
 		gint numerator, denominator;
 		if (gst_structure_get_fraction (structure, "framerate", &numerator, &denominator))
 		{
-			FILE *f = fopen("/proc/stb/vmpeg/0/fallback_framerate", "w");
+			FILE *f = fopen("/usr/local/e2/etc/stb/vmpeg/0/fallback_framerate", "w");
 			if (f)
 			{
 				int valid_framerates[] = { 23976, 24000, 25000, 29970, 30000, 50000, 59940, 60000 };
@@ -1730,7 +1730,7 @@
 
 	self->pesheader_buffer = gst_buffer_new_and_alloc(2048);
 
-	f = fopen("/proc/stb/vmpeg/0/fallback_framerate", "r");
+	f = fopen("/usr/local/e2/etc/stb/vmpeg/0/fallback_framerate", "r");
 	if (f)
 	{
 		fgets(self->saved_fallback_framerate, sizeof(self->saved_fallback_framerate), f);
@@ -1800,7 +1800,7 @@
 		queue_pop(&self->queue);
 	}
 
-	f = fopen("/proc/stb/vmpeg/0/fallback_framerate", "w");
+	f = fopen("/usr/local/e2/etc/stb/vmpeg/0/fallback_framerate", "w");
 	if (f)
 	{
 		fputs(self->saved_fallback_framerate, f);
