diff -ruN o/src/EPGImport/EPGConfig.py pc/src/EPGImport/EPGConfig.py
--- o/src/EPGImport/EPGConfig.py	2020-04-12 13:39:06.645411006 +0300
+++ pc/src/EPGImport/EPGConfig.py	2020-07-30 15:54:58.925071903 +0300
@@ -81,7 +81,7 @@
 			print>>log, "[EPGImport] failed to parse", downloadedFile, "Error:", e
 			pass
 	def update(self, filterCallback, downloadedFile=None):
-		customFile='/etc/epgimport/custom.channels.xml'
+		customFile='/usr/local/e2/etc/epgimport/custom.channels.xml'
 		# Always read custom file since we don't know when it was last updated
 		# and we don't have multiple download from server problem since it is always a local file.
 		if os.path.exists(customFile):
diff -ruN o/src/EPGImport/EPGImport.py pc/src/EPGImport/EPGImport.py
--- o/src/EPGImport/EPGImport.py	2020-04-12 13:39:06.645411006 +0300
+++ pc/src/EPGImport/EPGImport.py	2020-07-30 09:52:10.000000000 +0300
@@ -10,7 +10,7 @@
 import log
 import random
 
-HDD_EPG_DAT = "/hdd/epg.dat"
+HDD_EPG_DAT = "/media/hdd/epg.dat"
 
 from twisted.internet import reactor, threads
 from twisted.web.client import downloadPage
diff -ruN o/src/EPGImport/filtersServices.py pc/src/EPGImport/filtersServices.py
--- o/src/EPGImport/filtersServices.py	2020-04-12 13:39:06.645411006 +0300
+++ pc/src/EPGImport/filtersServices.py	2020-07-30 15:56:01.382010084 +0300
@@ -58,8 +58,8 @@
 
 	def saveTo(self, filename):
 		try:
-			if not os.path.isdir('/etc/epgimport'):
-				os.system('mkdir /etc/epgimport')
+			if not os.path.isdir('/usr/local/e2/etc/epgimport'):
+				os.system('mkdir /usr/local/e2/etc/epgimport')
 			cfg = open(filename, 'w')
 		except:
 			return
@@ -68,7 +68,7 @@
 		cfg.close()
 
 	def load(self):
-		self.loadFrom('/etc/epgimport/ignore.conf')
+		self.loadFrom('/usr/local/e2/etc/epgimport/ignore.conf')
 
 	def reload(self):
 		self.services = []
@@ -78,7 +78,7 @@
 		return self.services
 
 	def save(self):
-		self.saveTo('/etc/epgimport/ignore.conf')
+		self.saveTo('/usr/local/e2/etc/epgimport/ignore.conf')
 
 	def addService(self, ref):
 		if isinstance(ref, str):
diff -ruN o/src/EPGImport/locale/ru.po pc/src/EPGImport/locale/ru.po
--- o/src/EPGImport/locale/ru.po	2020-07-29 11:03:45.728171771 +0300
+++ pc/src/EPGImport/locale/ru.po	2020-07-30 16:04:08.406992202 +0300
@@ -199,7 +199,7 @@
 
 #: ../plugin.py:287
 msgid "Manual"
-msgstr "В ручную"
+msgstr "Вручную"
 
 #: ../plugin.py:80
 msgid "Monday"
diff -ruN o/src/EPGImport/plugin.py pc/src/EPGImport/plugin.py
--- o/src/EPGImport/plugin.py	2020-06-07 23:04:36.225164857 +0300
+++ pc/src/EPGImport/plugin.py	2020-07-30 15:56:37.787682597 +0300
@@ -94,7 +94,7 @@
 from Plugins.Plugin import PluginDescriptor
 
 # historically located (not a problem, we want to update it)
-CONFIG_PATH = '/etc/epgimport'
+CONFIG_PATH = '/usr/local/e2/etc/epgimport'
 
 # Global variable
 autoStartTimer = None
