diff -ruN o/plugin/controllers/ajax.py pc/plugin/controllers/ajax.py
--- o/plugin/controllers/ajax.py	2022-08-31 12:17:33.111415132 +0300
+++ pc/plugin/controllers/ajax.py	2022-08-31 12:18:54.314450976 +0300
@@ -178,7 +178,7 @@
 			box['brand'] = "iqon"
 		elif getMachineBrand() == 'Technomate':
 			box['brand'] = "techomate"
-		elif fileExists("/proc/stb/info/azmodel"):
+		elif fileExists("/usr/local/e2/etc/info/azmodel"):
 			box['brand'] = "azbox"
 
 		return {"box": box,
diff -ruN o/plugin/controllers/defaults.py pc/plugin/controllers/defaults.py
--- o/plugin/controllers/defaults.py	2022-08-31 12:17:33.112415081 +0300
+++ pc/plugin/controllers/defaults.py	2022-08-31 12:18:54.314450976 +0300
@@ -56,7 +56,7 @@
 
 
 def getTranscoding():
-	if os.path.isfile("/proc/stb/encoder/0/bitrate"):
+	if os.path.isfile("/usr/local/e2/etc/encoder/0/bitrate"):
 		return isPluginInstalled("TranscodingSetup") or isPluginInstalled("TransCodingSetup") or isPluginInstalled("MultiTransCodingSetup")
 	return False
 
diff -ruN o/plugin/controllers/models/control.py pc/plugin/controllers/models/control.py
--- o/plugin/controllers/models/control.py	2022-08-31 12:17:33.112415081 +0300
+++ pc/plugin/controllers/models/control.py	2022-08-31 12:18:54.314450976 +0300
@@ -30,7 +30,7 @@
 import os
 from six.moves.urllib.parse import unquote
 
-ENABLE_QPIP_PROCPATH = "/proc/stb/video/decodermode"
+ENABLE_QPIP_PROCPATH = "/usr/local/e2/etc/video/decodermode"
 
 try:
 	from enigma import setPrevAsciiCode
diff -ruN o/plugin/controllers/models/info.py pc/plugin/controllers/models/info.py
--- o/plugin/controllers/models/info.py	2022-08-31 12:17:33.112415081 +0300
+++ pc/plugin/controllers/models/info.py	2022-08-31 12:18:54.314450976 +0300
@@ -231,7 +231,7 @@
 		elif model in ("topf", "tf7700hdpvr"):
 			chipset = "SH4 @266MHz"
 		elif model.startswith("azbox"):
-			f = open("/proc/stb/info/model", 'r')
+			f = open("/usr/local/e2/etc/info/model", 'r')
 			model = f.readline().strip().lower()
 			f.close()
 			if model == "me":
@@ -245,8 +245,8 @@
 				chipset = "SH4 @540MHz"
 			else:
 				chipset = "SH4 @450MHz"
-	elif fileExists("/proc/stb/info/azmodel"):
-		f = open("/proc/stb/info/model", 'r')
+	elif fileExists("/usr/local/e2/etc/info/azmodel"):
+		f = open("/usr/local/e2/etc/info/model", 'r')
 		model = f.readline().strip().lower()
 		f.close()
 		if model == "me":
@@ -255,8 +255,8 @@
 			chipset = "SIGMA 8653"
 		else:
 			chipset = "SIGMA 8634"
-	elif fileExists("/proc/stb/info/model"):
-		f = open("/proc/stb/info/model", 'r')
+	elif fileExists("/usr/local/e2/etc/info/model"):
+		f = open("/usr/local/e2/etc/info/model", 'r')
 		model = f.readline().strip().lower()
 		f.close()
 		if model == "tf7700hdpvr":
@@ -287,8 +287,8 @@
 		elif model in ("dm900", "dm920"):
 			chipset = "bcm7252S"
 
-	if fileExists("/proc/stb/info/chipset"):
-		f = open("/proc/stb/info/chipset", 'r')
+	if fileExists("/usr/local/e2/etc/info/chipset"):
+		f = open("/usr/local/e2/etc/info/chipset", 'r')
 		chipset = f.readline().strip()
 		f.close()
 
diff -ruN o/plugin/controllers/models/locations.py pc/plugin/controllers/models/locations.py
--- o/plugin/controllers/models/locations.py	2022-08-31 12:17:33.112415081 +0300
+++ pc/plugin/controllers/models/locations.py	2022-08-31 12:18:54.315450929 +0300
@@ -20,9 +20,9 @@
 
 
 def getCurrentLocation():
-	path = config.movielist.last_videodir.value or "/hdd/movie"
+	path = config.movielist.last_videodir.value or "/media/hdd/movie"
 	if not os.path.exists(path):
-		path = "/hdd/movie"
+		path = "/media/hdd/movie"
 
 	return {
 		"result": True,
diff -ruN o/plugin/controllers/models/owibranding.py pc/plugin/controllers/models/owibranding.py
--- o/plugin/controllers/models/owibranding.py	2022-08-31 12:17:33.113415030 +0300
+++ pc/plugin/controllers/models/owibranding.py	2022-08-31 12:18:54.315450929 +0300
@@ -144,9 +144,9 @@
 	if tpmloaded:
 		orgdream = tpm_check()
 
-	if fileExists("/proc/stb/info/hwmodel"):
+	if fileExists("/usr/local/e2/etc/info/hwmodel"):
 		brand = "DAGS"
-		f = open("/proc/stb/info/hwmodel", 'r')
+		f = open("/usr/local/e2/etc/info/hwmodel", 'r')
 		procmodel = f.readline().strip()
 		f.close()
 		if (procmodel.startswith("optimuss") or procmodel.startswith("pingulux")):
@@ -176,15 +176,15 @@
 				model = procmodel.replace("lunix4k", "Lunix4K")
 			elif procmodel == "dual":
 				model = procmodel.replace("dual", "Dual")
-	elif fileExists("/proc/stb/info/azmodel"):
+	elif fileExists("/usr/local/e2/etc/info/azmodel"):
 		brand = "AZBox"
-		f = open("/proc/stb/info/model", 'r')  # To-Do: Check if "model" is really correct ...
+		f = open("/usr/local/e2/etc/info/model", 'r')  # To-Do: Check if "model" is really correct ...
 		procmodel = f.readline().strip()
 		f.close()
 		model = procmodel.lower()
-	elif fileExists("/proc/stb/info/gbmodel"):
+	elif fileExists("/usr/local/e2/etc/info/gbmodel"):
 		brand = "GigaBlue"
-		f = open("/proc/stb/info/gbmodel", 'r')
+		f = open("/usr/local/e2/etc/info/gbmodel", 'r')
 		procmodel = f.readline().strip()
 		f.close()
 		if procmodel == "GBQUAD PLUS":
@@ -199,9 +199,9 @@
 			model = procmodel.replace("ue4k", "UHD UE 4k")
 		elif procmodel == "gbtrio4k":
 			model = procmodel.replace("gbtrio4k", "UHD Trio 4k")
-	elif fileExists("/proc/stb/info/vumodel") and not fileExists("/proc/stb/info/boxtype"):
+	elif fileExists("/usr/local/e2/etc/info/vumodel") and not fileExists("/usr/local/e2/etc/info/boxtype"):
 		brand = "Vu+"
-		f = open("/proc/stb/info/vumodel", 'r')
+		f = open("/usr/local/e2/etc/info/vumodel", 'r')
 		procmodel = f.readline().strip()
 		f.close()
 		model = procmodel.title().replace("olose", "olo SE").replace("olo2se", "olo2 SE").replace("2", "²").replace("4Kse", "4K SE")
@@ -227,8 +227,8 @@
 				model = "UHD 88"
 			else:
 				model = "ESI 88"
-	elif fileExists("/proc/stb/info/boxtype"):
-		f = open("/proc/stb/info/boxtype", 'r')
+	elif fileExists("/usr/local/e2/etc/info/boxtype"):
+		f = open("/usr/local/e2/etc/info/boxtype", 'r')
 		procmodel = f.readline().strip().lower()
 		f.close()
 		if procmodel.startswith("et"):
@@ -528,7 +528,7 @@
 		elif procmodel.startswith("sf"):
 			brand = "Octagon"
 			if procmodel.startswith("sf8008"):
-				sf8008type = open("/proc/stb/info/type").read()
+				sf8008type = open("/usr/local/e2/etc/info/type").read()
 				if sf8008type.startswith("11"):
 					procmodel = "sf8008t"
 					model = "SF8008 4K Twin"
@@ -557,8 +557,8 @@
 		else:
 			model = procmodel
 
-	elif fileExists("/proc/stb/info/model"):
-		f = open("/proc/stb/info/model", 'r')
+	elif fileExists("/usr/local/e2/etc/info/model"):
+		f = open("/usr/local/e2/etc/info/model", 'r')
 		procmodel = f.readline().strip().lower()
 		f.close()
 		if procmodel == "tf7700hdpvr":
diff -ruN o/plugin/controllers/models/services.py pc/plugin/controllers/models/services.py
--- o/plugin/controllers/models/services.py	2022-08-31 12:17:33.113415030 +0300
+++ pc/plugin/controllers/models/services.py	2022-08-31 12:18:54.315450929 +0300
@@ -21,6 +21,9 @@
 ##########################################################################
 
 from datetime import datetime
+import sys
+reload(sys)
+sys.setdefaultencoding('utf8')
 import re
 import six
 from six.moves.urllib.parse import quote, unquote
diff -ruN o/plugin/controllers/models/stream.py pc/plugin/controllers/models/stream.py
--- o/plugin/controllers/models/stream.py	2022-08-31 12:17:33.113415030 +0300
+++ pc/plugin/controllers/models/stream.py	2022-08-31 12:18:54.315450929 +0300
@@ -90,10 +90,10 @@
 		_port = getUrlArg(request, "port")
 		if _port != None:
 			portNumber = _port
-	elif fileExists("/dev/encoder0") or fileExists("/proc/stb/encoder/0/apply"):
+	elif fileExists("/dev/encoder0") or fileExists("/usr/local/e2/etc/encoder/0/apply"):
 		transcoder_port = portNumber
 
-	if fileExists("/dev/bcm_enc0") or fileExists("/dev/encoder0") or fileExists("/proc/stb/encoder/0/apply"):
+	if fileExists("/dev/bcm_enc0") or fileExists("/dev/encoder0") or fileExists("/usr/local/e2/etc/encoder/0/apply"):
 		if device == "phone":
 			try:
 				bitrate = config.plugins.transcodingsetup.bitrate.value
@@ -102,7 +102,7 @@
 				# framerate = config.plugins.transcodingsetup.framerate.value
 				aspectratio = config.plugins.transcodingsetup.aspectratio.value
 				interlaced = config.plugins.transcodingsetup.interlaced.value
-				if fileExists("/proc/stb/encoder/0/vcodec"):
+				if fileExists("/usr/local/e2/etc/encoder/0/vcodec"):
 					vcodec = config.plugins.transcodingsetup.vcodec.value
 					args = "?bitrate=%s__width=%s__height=%s__vcodec=%s__aspectratio=%s__interlaced=%s" % (bitrate, width, height, vcodec, aspectratio, interlaced)
 				else:
@@ -187,7 +187,7 @@
 
 		device = getUrlArg(request, "device")
 
-		if fileExists("/dev/bcm_enc0") or fileExists("/dev/encoder0") or fileExists("/proc/stb/encoder/0/apply"):
+		if fileExists("/dev/bcm_enc0") or fileExists("/dev/encoder0") or fileExists("/usr/local/e2/etc/encoder/0/apply"):
 			try:
 				transcoder_port = int(config.plugins.transcodingsetup.port.value)
 			except Exception:
@@ -199,7 +199,7 @@
 			if _port != None:
 				portNumber = _port
 
-		if fileExists("/dev/bcm_enc0") or fileExists("/dev/encoder0") or fileExists("/proc/stb/encoder/0/apply"):
+		if fileExists("/dev/bcm_enc0") or fileExists("/dev/encoder0") or fileExists("/usr/local/e2/etc/encoder/0/apply"):
 			if device == "phone":
 				try:
 					bitrate = config.plugins.transcodingsetup.bitrate.value
@@ -208,7 +208,7 @@
 					# framerate = config.plugins.transcodingsetup.framerate.value
 					aspectratio = config.plugins.transcodingsetup.aspectratio.value
 					interlaced = config.plugins.transcodingsetup.interlaced.value
-					if fileExists("/proc/stb/encoder/0/vcodec"):
+					if fileExists("/usr/local/e2/etc/encoder/0/vcodec"):
 						vcodec = config.plugins.transcodingsetup.vcodec.value
 						args = "?bitrate=%s__width=%s__height=%s__vcodec=%s__aspectratio=%s__interlaced=%s" % (bitrate, width, height, vcodec, aspectratio, interlaced)
 					else:
diff -ruN o/plugin/controllers/models/timers.py pc/plugin/controllers/models/timers.py
--- o/plugin/controllers/models/timers.py	2022-08-31 12:17:33.113415030 +0300
+++ pc/plugin/controllers/models/timers.py	2022-08-31 12:18:54.315450929 +0300
@@ -675,7 +675,7 @@
 		location = request.args['dirname'][0]
 
 	if not location:
-		location = "/hdd/movie/"
+		location = "/media/hdd/movie/"
 
 	begin = int(mktime((int(request.args['syear'][0]), int(request.args['smonth'][0]), int(request.args['sday'][0]), int(request.args['shour'][0]), int(request.args['smin'][0]), 0, 0, 0, -1)))
 	end = int(mktime((int(request.args['syear'][0]), int(request.args['smonth'][0]), int(request.args['sday'][0]), int(request.args['ehour'][0]), int(request.args['emin'][0]), 0, 0, 0, -1)))
diff -ruN o/testsuite/evil_eval.py pc/testsuite/evil_eval.py
--- o/testsuite/evil_eval.py	2022-08-31 12:17:33.254407808 +0300
+++ pc/testsuite/evil_eval.py	2022-08-31 12:18:54.316450883 +0300
@@ -9,7 +9,7 @@
 
     root@heart-of-gold:~# curl --noproxy localhost http://localhost/api/saveconfig --data "value=1&key=config.__class__.__name__ == 1 or (open('you lost your mind? according to my last psych EVALuation, yes', 'wb') and config or config)"
     root@heart-of-gold:~# ls $HOME/you*
-    /home/root/you lost your mind? according to my last psych EVALuation, yes
+    /media/hdd/you lost your mind? according to my last psych EVALuation, yes
 
     root@heart-of-gold:~# curl --noproxy localhost http://localhost/api/saveconfig --data "value=1&key=config.__class__ == 42 or (__import__('os').system('id > pwned') and config or config)"
     {"result": false}
diff -ruN o/testsuite/status_quo_file_controller.py pc/testsuite/status_quo_file_controller.py
--- o/testsuite/status_quo_file_controller.py	2022-08-31 12:17:33.254407808 +0300
+++ pc/testsuite/status_quo_file_controller.py	2022-08-31 12:18:54.316450883 +0300
@@ -71,7 +71,7 @@
 		}
 		req = requests.get(self.file_url, params=params)
 		print("Tried to fetch {!r}".format(req.url))
-		self.assertEqual("File '/home/root/{:s}' not found".format(randy),
+		self.assertEqual("File '/media/hdd/{:s}' not found".format(randy),
 						  req.text)
 		self.assertEqual(200, req.status_code)
 
