diff -ruN o/plugin/controllers/models/locations.py pc/plugin/controllers/models/locations.py
--- o/plugin/controllers/models/locations.py	2022-08-26 12:09:43.904955520 +0300
+++ pc/plugin/controllers/models/locations.py	2022-08-26 12:11:22.782322152 +0300
@@ -20,9 +20,9 @@
 
 
 def getCurrentLocation():
-	path = config.movielist.last_videodir.value or "/hdd/movie"
+	path = config.movielist.last_videodir.value or "/media/hdd/movie"
 	if not os.path.exists(path):
-		path = "/hdd/movie"
+		path = "/media/hdd/movie"
 
 	return {
 		"result": True,
diff -ruN o/plugin/controllers/models/services.py pc/plugin/controllers/models/services.py
--- o/plugin/controllers/models/services.py	2022-08-26 12:09:43.908955577 +0300
+++ pc/plugin/controllers/models/services.py	2022-08-26 12:11:22.782322152 +0300
@@ -21,6 +21,9 @@
 ##########################################################################
 
 from datetime import datetime
+import sys
+reload(sys)
+sys.setdefaultencoding('utf8')
 import re
 import six
 from six.moves.urllib.parse import quote, unquote
diff -ruN o/plugin/controllers/models/timers.py pc/plugin/controllers/models/timers.py
--- o/plugin/controllers/models/timers.py	2022-08-26 12:09:43.908955577 +0300
+++ pc/plugin/controllers/models/timers.py	2022-08-26 12:11:22.782322152 +0300
@@ -675,7 +675,7 @@
 		location = request.args['dirname'][0]
 
 	if not location:
-		location = "/hdd/movie/"
+		location = "/media/hdd/movie/"
 
 	begin = int(mktime((int(request.args['syear'][0]), int(request.args['smonth'][0]), int(request.args['sday'][0]), int(request.args['shour'][0]), int(request.args['smin'][0]), 0, 0, 0, -1)))
 	end = int(mktime((int(request.args['syear'][0]), int(request.args['smonth'][0]), int(request.args['sday'][0]), int(request.args['ehour'][0]), int(request.args['emin'][0]), 0, 0, 0, -1)))
diff -ruN o/testsuite/evil_eval.py pc/testsuite/evil_eval.py
--- o/testsuite/evil_eval.py	2022-08-26 12:09:44.152959052 +0300
+++ pc/testsuite/evil_eval.py	2022-08-26 12:11:22.782322152 +0300
@@ -9,7 +9,7 @@
 
     root@heart-of-gold:~# curl --noproxy localhost http://localhost/api/saveconfig --data "value=1&key=config.__class__.__name__ == 1 or (open('you lost your mind? according to my last psych EVALuation, yes', 'wb') and config or config)"
     root@heart-of-gold:~# ls $HOME/you*
-    /home/root/you lost your mind? according to my last psych EVALuation, yes
+    /media/hdd/you lost your mind? according to my last psych EVALuation, yes
 
     root@heart-of-gold:~# curl --noproxy localhost http://localhost/api/saveconfig --data "value=1&key=config.__class__ == 42 or (__import__('os').system('id > pwned') and config or config)"
     {"result": false}
diff -ruN o/testsuite/status_quo_file_controller.py pc/testsuite/status_quo_file_controller.py
--- o/testsuite/status_quo_file_controller.py	2022-08-26 12:09:44.152959052 +0300
+++ pc/testsuite/status_quo_file_controller.py	2022-08-26 12:11:22.786322205 +0300
@@ -71,7 +71,7 @@
 		}
 		req = requests.get(self.file_url, params=params)
 		print("Tried to fetch {!r}".format(req.url))
-		self.assertEqual("File '/home/root/{:s}' not found".format(randy),
+		self.assertEqual("File '/media/hdd/{:s}' not found".format(randy),
 						  req.text)
 		self.assertEqual(200, req.status_code)
 
