diff -ruN b/lib/dvb/frontend.cpp a/lib/dvb/frontend.cpp
--- b/lib/dvb/frontend.cpp	2017-08-06 19:37:26.000000000 +0200
+++ a/lib/dvb/frontend.cpp	2017-08-06 19:34:14.000000000 +0200
@@ -50,6 +50,18 @@
 		}
 #endif
 
+template<class T>
+const T& constrain(const T& x, const T& a, const T& b) {
+    if(x < a) {
+        return a;
+    }
+    else if(b < x) {
+        return b;
+    }
+    else
+        return x;
+}
+
 void eDVBDiseqcCommand::setCommandString(const char *str)
 {
 	if (!str)
@@ -1167,6 +1179,37 @@
 				break;
 		}
 	}
+	else if (strstr(m_description, "TechniSat SkyStar HD2"))
+	{
+		int MinSnr = 10;
+		int MaxSnr = 670;
+
+		eDVBFrontendParametersSatellite parm;
+		oparm.getDVBS(parm);
+
+		switch (parm.system)
+		{
+			case eDVBFrontendParametersSatellite::System_DVB_S:
+				MaxSnr = 200;
+				break;
+			case eDVBFrontendParametersSatellite::System_DVB_S2:
+				MaxSnr = 70;
+			default: break;
+		}
+
+		int Ber = readFrontendData(iFrontendInformation_ENUMS::bitErrorRate);
+
+		int a = int(constrain(snr, MinSnr, MaxSnr)) * 100 / (MaxSnr - MinSnr);
+		int b = 100 - ((Ber / 256) * 5);
+		if (b < 0)
+			b = 0;
+		int q = 15 + a * b * 95 / 100 / 100;
+		if (q > 100)
+			q = 100;
+
+		ret = q;
+		sat_max = MaxSnr;
+	}
 
 	signalqualitydb = ret;
 	if (ret == 0x12345678) // no snr db calculation avail.. return untouched snr value..
