diff -ruN o/libtuxtxt/acinclude.m4 pc/libtuxtxt/acinclude.m4
--- o/libtuxtxt/acinclude.m4	2021-06-20 20:03:04.000000000 +0300
+++ pc/libtuxtxt/acinclude.m4	2021-08-02 16:47:20.385638489 +0300
@@ -189,7 +189,7 @@
 #include <linux/dvb/version.h>
 version DVB_API_VERSION
 	]])])
-	DVB_API_VERSION=`(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
+	DVB_API_VERSION=`(eval "$ac_cpp -traditional-cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
 	rm -f conftest*
 
 	AC_MSG_NOTICE([found dvb version $DVB_API_VERSION])
diff -ruN o/libtuxtxt/tuxtxt_common.h pc/libtuxtxt/tuxtxt_common.h
--- o/libtuxtxt/tuxtxt_common.h	2021-06-20 20:03:04.000000000 +0300
+++ pc/libtuxtxt/tuxtxt_common.h	2021-08-02 16:47:20.387638398 +0300
@@ -4061,14 +4061,6 @@
 *************************************************************************/
 
 /* devices */
-#define AVS "/dev/dbox/avs0"
-#define SAA "/dev/dbox/saa0"
-#if HAVE_DVB_API_VERSION < 3
-#define PIG "/dev/dbox/pig0"
-#else
-#define PIG "/dev/v4l/video0"
-#endif
-
 #define TOPMENUSTARTX TV43STARTX+2
 #define TOPMENUENDX TVENDX
 #define TOPMENUSTARTY renderinfo->StartY
diff -ruN o/tuxtxt/acinclude.m4 pc/tuxtxt/acinclude.m4
--- o/tuxtxt/acinclude.m4	2021-06-20 20:03:04.000000000 +0300
+++ pc/tuxtxt/acinclude.m4	2021-08-02 16:47:20.387638398 +0300
@@ -189,7 +189,7 @@
 #include <linux/dvb/version.h>
 version DVB_API_VERSION
 	]])])
-	DVB_API_VERSION=`(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
+	DVB_API_VERSION=`(eval "$ac_cpp -traditional-cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
 	rm -f conftest*
 
 	AC_MSG_NOTICE([found dvb version $DVB_API_VERSION])
diff -ruN o/tuxtxt/data/Makefile.am pc/tuxtxt/data/Makefile.am
--- o/tuxtxt/data/Makefile.am	2021-06-20 20:03:04.000000000 +0300
+++ pc/tuxtxt/data/Makefile.am	2021-08-02 16:47:20.387638398 +0300
@@ -1,4 +1,4 @@
-installdir = $(sysconfdir)/tuxtxt/
+installdir = /usr/local/e2/tuxtxt/
 
 install_DATA = \
 	tuxtxt2.conf
diff -ruN o/tuxtxt/tuxtxt.h pc/tuxtxt/tuxtxt.h
--- o/tuxtxt/tuxtxt.h	2021-06-20 20:03:04.000000000 +0300
+++ pc/tuxtxt/tuxtxt.h	2021-08-02 16:47:20.387638398 +0300
@@ -94,17 +94,13 @@
 #endif
 
 #ifndef FB_DEV
-# define FB_DEV "/dev/fb/0"
+# define FB_DEV "/dev/fb0"
 #endif
 
-#define TUXTXTCONF CONFIGDIR "/tuxtxt/tuxtxt2.conf"
-
+#define TUXTXTCONF CONFIGDIR "/usr/local/e2/tuxtxt/tuxtxt2.conf"
 
 #define fontwidth_small_lcd 8
 
-
-
-
 /* rc codes */
 #if HAVE_DVB_API_VERSION < 3
 #define KEY_0       0x5C00
