diff -ruN o/oscam-patched/oscam.c n/oscam-patched/oscam.c
--- o/oscam-patched/oscam.c	2019-07-31 23:36:45.656521579 +0300
+++ n/oscam-patched/oscam.c	2019-07-31 23:35:28.974149974 +0300
@@ -850,10 +850,10 @@
 	int8_t azmodel = 0;
 	FILE *f;
 
-	if ((f = fopen("/proc/stb/info/azmodel", "r"))){ azmodel = 1; fclose(f);}
-	read_line_from_file("/proc/stb/info/model", model, sizeof(model));
-	read_line_from_file("/proc/stb/info/boxtype", boxtype, sizeof(boxtype));
-	read_line_from_file("/proc/stb/info/vumodel", vumodel, sizeof(vumodel));
+	if ((f = fopen("/usr/local/e2/etc/stb/info/azmodel", "r"))){ azmodel = 1; fclose(f);}
+	read_line_from_file("/usr/local/e2/etc/stb/info/model", model, sizeof(model));
+	read_line_from_file("/usr/local/e2/etc/stb/info/boxtype", boxtype, sizeof(boxtype));
+	read_line_from_file("/usr/local/e2/etc/stb/info/vumodel", vumodel, sizeof(vumodel));
 	if (vumodel[0] && !boxtype[0] && !azmodel)
 	{
 		snprintf(boxtype, sizeof(boxtype), "vu%s", vumodel);
